tests/suites/os/tests/secureboot/index.js

```diff
@@ -42,25 +53,64 @@ class secureBoot {
 	}
 
 	async testFullDiskEncryption() {
-			Promise.all(
-				[
-					{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-					{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-					{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-					{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-				].map(args => this.test.resolveMatch(
-						this.worker.executeCommandInHostOS(
-							[
-								'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-								'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-							],
-							this.link,
-						),
-						/crypt/,
-						`${args.name} is encrypted`,
-					)
+		await Promise.all(
+			[
+				{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
+				{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
+				{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
+				{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
+			].map(args => this.test.resolveMatch(
+					this.worker.executeCommandInHostOS(
+						[
+							'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
+							'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
+						],
+						this.link,
+					),
+					/crypt/,
```

Comment:
This check will only work for LUKS devices - we will have to re-do it soon to use dmsetup instead to detect dm-crypt encrypted partitions in general. There is an example bash implementation in https://github.com/balena-os/meta-balena/pull/3523/commits/34781196c7dc3beaab866164fed84408ea678847